components/registry/org.wso2.carbon.registry.extensions/src/main/java/org/wso2/carbon/registry/extensions/handlers/utils/RESTServiceUtils.java

```diff
@@ -0,0 +1,478 @@
+		String serviceVersion =
+				overview.getFirstChildWithName(new QName(CommonConstants.SERVICE_ELEMENT_NAMESPACE, VERSION)).getText();
+		String apiName =
+				overview.getFirstChildWithName(new QName(CommonConstants.SERVICE_ELEMENT_NAMESPACE, NAME)).getText();
+		serviceVersion = (serviceVersion == null) ? CommonConstants.SERVICE_VERSION_DEFAULT_VALUE : serviceVersion;
+
+		//set version property.
+		serviceResource.setProperty(RegistryConstants.VERSION_PARAMETER_NAME, serviceVersion);
+		//set content.
+		serviceResource.setContent(RegistryUtils.encodeString(data.toString()));
+
+		String resourceId = serviceResource.getUUID();
+		//set resource UUID
+		resourceId = (resourceId == null) ? UUID.randomUUID().toString() : resourceId;
+
+		serviceResource.setUUID(resourceId);
+		String servicePath = getChrootedServiceLocation(requestContext.getRegistryContext()) +
+		                     CarbonContext.getThreadLocalCarbonContext().getUsername() +
+		                     RegistryConstants.PATH_SEPARATOR + apiName +
+		                     RegistryConstants.PATH_SEPARATOR + serviceVersion +
+		                     RegistryConstants.PATH_SEPARATOR + apiName + "-rest_service";
+		//saving the api resource to repository.
+		registry.put(servicePath, serviceResource);
+		log.info("REST Service created at " + servicePath);
+		return servicePath;
+	}
+
+	/**
+	 * Adds the service endpoint element to the registry.
+	 *
+	 * @param requestContext        current request information.
+	 * @param endpointElement       endpoint metadata element.
+	 * @param endpointPath          endpoint location.
+	 * @return                      The resource path of the endpoint.
+	 * @throws RegistryException    If fails to add the endpoint to the registry.
+	 */
+	public static String addEndpointToRegistry(RequestContext requestContext, OMElement endpointElement, String endpointPath)
+			throws RegistryException {
+
+		if(requestContext == null || endpointElement == null || endpointPath == null) {
+			throw new IllegalArgumentException("Some or all of the arguments may be null. Cannot add the endpoint to registry. ");
+		}
+
+		Registry registry = requestContext.getRegistry();
+		//Creating new resource.
+		Resource endpointResource = new ResourceImpl();
+		//setting endpoint media type.
+		endpointResource.setMediaType(CommonConstants.ENDPOINT_MEDIA_TYPE);
+		//set content.
+		endpointResource.setContent(RegistryUtils.encodeString(endpointElement.toString()));
+		//set path
+		endpointPath = getChrootedEndpointLocation(requestContext.getRegistryContext()) + endpointPath;
+
+		String resourceId = endpointResource.getUUID();
+		//set resource UUID
+		resourceId = (resourceId == null) ? UUID.randomUUID().toString() : resourceId;
+
+		endpointResource.setUUID(resourceId);
+		//saving the api resource to repository.
+		registry.put(endpointPath, endpointResource);
+		log.info("Endpoint created at " + endpointPath);
```

Comment:
@prasa7 This info log is already converted to a debug log in 2a77e8a7f0d0bfd6157db1e30ac8783b9eccd3f8